### DISCUSSION TITLE
A question for the PowerShell developers.

### DISCUSSION THREAD

**Discussion by kris3713** [PowerShell/PowerShell#20341] [Q&A] [answered]
I just want to know if the developer team for PowerShell used any external libraries for parsing CSV files. Or, did the Dev team create their own libraries for parsing CSV files?

Also, I'm a beginner developer, and I am trying to learn how to parse a CSV more effectively. I want an alternative from using the `TextFieldParser` method that originates from the namespace `Microsoft.VisualBasic.FileIO`.

### EDIT
This is the CSV file I am trying to parse. 
[data.csv](https://github.com/PowerShell/PowerShell/files/12710867/data.csv)

**Comment by iSazonov** [comment, marked as answer]
pwsh uses its own parser. You can use follow cmdlets:
```powershell
gcm *-CSV*

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Cmdlet          ConvertFrom-Csv                                    7.0.0.0    Microsoft.PowerShell.Utility
Cmdlet          ConvertTo-Csv                                      7.0.0.0    Microsoft.PowerShell.Utility
Cmdlet          Export-Csv                                         7.0.0.0    Microsoft.PowerShell.Utility
Cmdlet          Import-Csv                                         7.0.0.0    Microsoft.PowerShell.Utility
```
